### PR TITLE
First set of @internal doc definitions

### PR DESCRIPTION
**Pull request by distantnative**

### CODE REVIEW

File: src/Http/Header.php [hunk 1]
```diff
@@ -5,13 +5,15 @@
 use Kirby\Toolkit\F;
 
 /**
+ * The Header class provides methods
+ * for sending HTTP headers.
+ *
  * Makes sending HTTP headers a breeze
```
Comment (bastianallgeier) [inline src/Http/Header.php]: Duplicate

File: src/Cms/App.php [hunk 2]
```diff
@@ -169,6 +177,7 @@ public function api(): Api
 
     /**
      *  Apply a hook to the given value
+     * @internal
```
Comment (bastianallgeier) [inline src/Cms/App.php]: Referring to my comment above: why is the collection method internal for example?

File: src/Cms/Files.php [hunk 3]
```diff
@@ -3,10 +3,22 @@
 namespace Kirby\Cms;
 
 /**
+ * The `$files` object extends the general
+ * `Collection` class and refers to a
+ * collection of files, i.e. images, documents
+ * etc. Files can be filtered, searched,
+ * converted, modified or evaluated with the
+ * following methods:
```
Comment (bastianallgeier) [inline src/Cms/Files.php:11]: I think we shouldn't do this. It's too much targeted at the online docs and no longer works when you read it in the code. Those blocks need to work in both ways or we need to create specific text blocks for our website if it's not possible in some cases. But we shouldn't be messing up our doc blocks.

**Review by bastianallgeier — COMMENTED** [review]
I see a lot of @internal comments and it's really hard to review them here because the functions all get cut. They sometimes don't seem really internal to me. How should we proceed with them?

File: src/Cms/App.php (hunk 2 above)
Comment (distantnative) [reply src/Cms/App.php]: That's a mistake

File: src/Cms/Files.php (hunk 3 above)
Comment (distantnative) [reply src/Cms/Files.php:11]: In what other context than the one described would this class ever be used? Imo it still works when you read the code. And it saves us in many places to maintain very similiar descriptions in 2 places.

File: src/Cms/App.php (hunk 2 above)
Comment (distantnative) [reply src/Cms/App.php]: @bastianallgeier In general, I just went through and marked the methods I thought are more internal/very advanced developers. Feel free to change as you prefer

**Review by bastianallgeier — APPROVED** [review]

**Comment by bastianallgeier** [comment]
Finally managed to review everything and merge it :)

**Pull request merged**